x-pack/plugins/security/server/routes/authentication/common.ts

```diff
@@ -77,52 +77,61 @@ export function defineCommonRoutes({ router, authc, basePath, logger }: RouteDef
     );
   }
 
-  function getLoginAttemptForProviderType(providerType: string, next: string) {
+  function getLoginAttemptForProviderType(providerType: string, redirectURL: string) {
+    const [redirectURLPath] = redirectURL.split('#');
+    const redirectURLFragment =
+      redirectURL.length > redirectURLPath.length
+        ? redirectURL.substring(redirectURLPath.length)
+        : '';
+
     if (providerType === SAMLAuthenticationProvider.type) {
-      return { type: SAMLLogin.LoginInitiatedByUser, redirectURL: next };
+      return { type: SAMLLogin.LoginInitiatedByUser, redirectURLPath, redirectURLFragment };
     }
 
     if (providerType === OIDCAuthenticationProvider.type) {
-      return { type: OIDCLogin.LoginInitiatedByUser, redirectURL: next };
+      return { type: OIDCLogin.LoginInitiatedByUser, redirectURLPath };
```

Comment:
Well, yeah, I see we never specifically handled URL fragments in OIDC flow. I _believe_ the assumption was that since OIDC callback endpoint is exposed as `GET` (unlike SAML) URL fragment will be automatically preserved during a set of browser redirects. I just went ahead and double checked with Google and Auth0 as Identity Providers and, unfortunately, fragment is lost somewhere in the IdP land :crying_cat_face:

Reply:
Another problem for another day then :)